lib/container/rabbitmq_container.ex

```diff
@@ -0,0 +1,259 @@
+  end
+
+  @doc """
+  Overrides the default virtual host used for the RabbitMQ container.
+
+  ## Examples
+    
+    iex> config = RabbitMQContainer.new() |> RabbitMQContainer.with_virtual_host("/")
+    iex> config.password
+    "/"
+  """
+  def with_virtual_host(%__MODULE__{} = config, virtual_host) when is_binary(virtual_host) do
+    %{config | virtual_host: virtual_host}
+  end
+
+  @doc """
+  Overrides the default command used for the RabbitMQ container.
+
+  ## Examples
+    
+    iex> config = RabbitMQContainer.new() |> RabbitMQContainer.with_cmd(["sh", "-c", "rabbitmq-server"])
+    iex> config.cmd
+    ["sh", "-c", "rabbitmq-server"]
+  """
+  def with_cmd(%__MODULE__{} = config, cmd) when is_list(cmd) do
+    %{config | cmd: cmd}
+  end
+
+  @doc """
+  Retrieves the default Docker image for the RabbitMQ container
+  """
+  def default_image, do: @default_image
+
+  @doc """
+  Retrieves the default exposed port for the RabbitMQ container
+  """
+  def default_port, do: @default_port
+
+  @doc """
+  Retrieves the default Docker image including tag for the RabbitMQ container
+  """
+  def default_image_with_tag, do: @default_image <> ":" <> @default_tag
+
+  @doc """
+  Returns the port on the _host machine_ where the RabbitMQ container is listening.
+  """
+  def port(%Container{} = container), do: Container.mapped_port(container, @default_port)
+
+  @doc """
+  Generates the connection URL for accessing the RabbitMQ service running within the container.
+
+  This URL is based on the AMQP 0-9-1.
+
+  ## Parameters
+
+  - `container`: The active RabbitMQ container instance in the form of a %Container{} struct.
+
+  ## Examples
+
+      iex> RabbitMQContainer.connection_url(container)
+      "amqp://guest:guest@localhost:32768" 
+      iex> RabbitMQContainer.connection_url(container_with_vhost)
+      "amqp://guest:guest@localhost:32768/vhost" 
+  """
+  def connection_url(%Container{} = container) do
+    "amqp://#{container.environment[:RABBITMQ_DEFAULT_USER]}:#{container.environment[:RABBITMQ_DEFAULT_PASS]}@#{Testcontainers.get_host()}:#{port(container)}#{virtual_host_segment(container)}"
+  end
+
+  @doc """
+  Returns the connection parameters to connect to RabbitMQ from the _host machine_.
+
+  ## Parameters
+
+  - `container`: The active RabbitMQ container instance in the form of a %Container{} struct.
+
+  ## Examples
+
+      iex> RabbitMQContainer.connection_parameters(container)
+      [
+        host: "localhost",
+        port: 32768,
+        username: "guest",
+        password: "guest",
+        vhost: "/"
+      ]
+  """
+  def connection_parameters(%Container{} = container) do
+    [
+      host: Testcontainers.get_host(),
+      port: port(container),
+      username: container.environment[:RABBITMQ_DEFAULT_USER],
+      password: container.environment[:RABBITMQ_DEFAULT_PASS],
+      virtual_host: container.environment[:RABBITMQ_DEFAULT_VHOST]
+    ]
+  end
+
+  defp virtual_host_segment(container) do
```

Comment:
could probably need some documentation about what it is.

Reply:
i mean i can see where its used but thats it

Reply:
I have provided docs detailing what this function is used for and how it is used in the AMQP URI scheme. (Relevant docs: https://www.rabbitmq.com/docs/uri-spec)